networkx/algorithms/link_analysis/hits_alg.py

```diff
@@ -77,11 +77,14 @@ def hits(G, max_iter=100, tol=1.0e-8, nstart=None, normalized=True):
         return {}, {}
     A = nx.adjacency_matrix(G, nodelist=list(G), dtype=float)
 
-    if nstart is None:
-        _, _, vt = sp.sparse.linalg.svds(A, k=1, maxiter=max_iter, tol=tol)
-    else:
+    if nstart is not None:
         nstart = np.array(list(nstart.values()))
+    if max_iter <= 0:
+        raise nx.PowerIterationFailedConvergence(max_iter)
```

Comment:
Just a nit, but I think a `ValueError` may be better here. "Practicality over Purity" though, so if this is a huge pain for backends then not a blocker!

Reply:
`PowerIterationFailedConvergence` was chosen here to match `pagerank`, `eigenvector_centrality`, and `katz_centrality`. I agree `ValueError` can make sense (esp. for `max_iter < 0`), but I would want to update the other algorithms too.

Reply:
+1 for consistency then, thanks!